engine/class_modules/sc_death_knight.cpp

```diff
@@ -4364,6 +4361,13 @@ struct death_coil_t : public death_knight_spell_t
     p() -> cooldown.dark_transformation -> adjust( -timespan_t::from_seconds(
       p() -> spec.death_coil -> effectN( 2 ).base_value() ) );
 
+    // Reduces the cooldown on Apocalypse by 1.5s
+    if ( p() -> spec.death_coil_2 -> ok() )
+    {
+      p()->cooldown.apocalypse->adjust(
+        -timespan_t::from_seconds( p() -> spec.death_coil -> effectN( 2 ).base_value() / 10) );
```

Comment:
I think this should be / 1000 instead of / 10

While we are here at this section, we may want to patch the cooldown for dark_transformation from r1 as well, and apply it to apocalypse instead.  Though both of these may change this week when they revert the dark transformation merge.

As a final note, does rank 1 and rank 2 of death coil both apply a CD reduction?  It looks like in most spells things such as this were moved to the second spell, not duplicated to it.

Reply:
if it is / 1000.0 then it is likely milliseconds and not seconds, and you can just use effectN( 2 ).time_value()

Reply:
For whatever reason the cd reduction value is stored in the spell data for rank 1, not rank 2.  The spell data for Rank 2 just has a single blank dummy effect.  I'll touch Apoc and DT in a week or two after Blizz maybe figures out what they're doing with those spells.

Reply:
We have seen that for many other rank2 spells as well. Quite a lot of the spell data for new stuff is horrible in shadowlands, in various ways.

But I think we just have to deal with it.